packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/e2e/quanticDocumentSuggestion.e2e.ts

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable no-await-in-loop */
+import {test, expect} from './fixture';
+
+const fakeTrackingId = 'tracking_id_123';
+
+['legacy', 'next'].forEach((mode) => {
+  test.describe(`quantic document suggestion ${mode} analytics`, () => {
+    let analyticsMode = 'legacy';
+    test.beforeEach(async ({documentSuggestion, page}) => {
+      analyticsMode = mode;
+      if (analyticsMode === 'next') {
+        await page.context().addCookies([
+          {
+            name: 'Coveo-Pendragon',
+            value: fakeTrackingId,
+            domain: '.my.site.com',
+            path: '/',
+          },
+          {
+            name: 'LSKey-c$Coveo-Pendragon',
+            value: fakeTrackingId,
+            domain: '.my.site.com',
+            path: '/',
+          },
+        ]);
+      }
+
+      await expect(documentSuggestion.noSuggestionsMessage).toBeVisible();
+    });
+
+    test('should render the component and all parts', async ({
+      documentSuggestion,
+      caseAssist,
+    }) => {
+      await expect(documentSuggestion.accordion).not.toBeVisible();
+      await expect(documentSuggestion.noSuggestionsMessage).toBeVisible();
+
+      const responsePromise = caseAssist.waitForDocumentSuggestionResponse();
+      await caseAssist.fetchSuggestions();
+      const response = await responsePromise;
+      const responseBody = await response.json();
+
+      await expect(documentSuggestion.accordion).toBeVisible();
+      await expect(documentSuggestion.noSuggestionsMessage).not.toBeVisible();
+      await expect(documentSuggestion.sections).toHaveCount(3);
+
+      for (const [index, section] of (
+        await documentSuggestion.sections.all()
+      ).entries()) {
+        await expect(section).toBeVisible();
+        await expect(section).toHaveText(
+          new RegExp(responseBody.documents[index].title)
+        );
+      }
+    });
+
+    test('should handle clicking on a document suggestion', async ({
+      page,
+      documentSuggestion,
+      caseAssist,
+    }) => {
+      const responsePromise = caseAssist.waitForDocumentSuggestionResponse();
+      await caseAssist.fetchSuggestions();
+      const response = await responsePromise;
+      const suggestions = await response.json();
+
+      // Hack(?) to have the request payload in the analytics request next.
+      await page.route('*analytics*', (route) => {
+        route.continue();
+      });
```

Comment:
ugh? not sure I understand why this is needed

Reply:
Check le commentaire! J'ai aucune idée pourquoi, mais c'est nécessaire pour avoir le payload de la request, sinon le payload est null :/